### PR TITLE
Fix Reset password link redirection issue (#1887)

### LINKED ISSUE #1887: Reset password link redirection issue when the user is signed in the TodoTemplate project
When a user use forgot password, we send her a reset password link to choose a new password.
If the user clicks on the reset password link again after she is signed in, the reset password page renders.
In that case, the user has to redirect to the last URL she visited or a friendly error message should be displayed.
image

### PR DESCRIPTION
**Pull request by mohabbati**
This closes #1887

### CODE REVIEW

**Comment by msynk** [comment]
@ysmoradi & @maryamhdr & @Cyrus-Sushiant plz review this PR

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/ResetPassword.razor.cs [hunk 1]
```diff
@@ -26,6 +26,9 @@ public partial class ResetPassword
 
     [Inject] public ITodoTemplateAuthenticationService TodoTemplateAuthenticationService { get; set; } = default!;
 
+    [Inject]
+    public TodoTemplateAuthenticationStateProvider TodoTemplateAuthenticationStateProvider { get; set; } = default!;
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/ResetPassword.razor.cs:30]: plz put the attribute in the same line as the property

File: src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/ResetPassword.razor.cs [hunk 2]
```diff
@@ -66,4 +69,15 @@ private async Task Submit()
             IsLoading = false;
         }
     }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            if (await TodoTemplateAuthenticationStateProvider.IsUserAuthenticated())
+                NavigationManager.NavigateTo("/");
```
Comment (msynk) [inline src/Tooling/Bit.Tooling.Templates/TodoTemplate/Web/Pages/ResetPassword.razor.cs:78]: plz use `{` and `}` here

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by ysmoradi** [comment]
We're using the same approach in sign in & up pages
@msynk

**Review by maryamhdr — APPROVED** [review]

**Review by msynk — APPROVED** [review]

**Pull request merged**